ResearchKit/Consent/ORKVisualConsentStepViewController.m

```diff
@@ -251,10 +252,20 @@ - (IBAction)goToPreviousPage {
 }
 
 - (IBAction)next {
-    [self showViewController:[self viewControllerForIndex:[self currentIndex]+1] forward:YES animated:YES];
-    ORKAccessibilityPostNotificationAfterDelay(UIAccessibilityScreenChangedNotification, nil, 0.5);
+    ORKConsentSceneViewController *currentConsentSceneViewController = [self viewControllerForIndex:[self currentIndex]];
+    [currentConsentSceneViewController setScrollEnabled:NO];
+    [currentConsentSceneViewController scrollToTopAnimated:YES completion:^(BOOL finished) {
+        [self showNextViewController];
```

Comment:
Upon testing, it seems that pressing *Cancel* doesn't interrupt the animation+transition, it just continues under the alert view. Anyway, I implemented what you suggested, just to be safe.